index.php

```diff
@@ -40,79 +39,76 @@
 
 $PAGE->set_context($personalcontext);
 
+$capabilities = array('report/coursecompletion:view',
+    'report/coursecompletion:view');
 $access = false;
-if (has_capability('report/coursecompletion:viewall', $systemcontext)) {
-    // User must be site admin or manager - can see records for all users
-    $access = true;
-} else if (has_capability('report/coursecompletion:view', $personalcontext)) {
-    // User is likely a parent/mentor - can see the student's records only
-    $access = true;
-} else if (has_capability('report/coursecompletion:view', $context)) {
-    // User must be a student - can see the logged in user's records only
-    $access = true;
-} else if (has_capability('report/coursecompletion:view', $systemcontext)) {
-    // User must be authenticated - can see the logged in user's records only
+
+if (has_any_capability( $capabilities, $systemcontext)) {
```

Comment:
The capability checking in here doesn't look quite right and it looks like you've removed the ability for the "parent" role to view a "students" records, but it doesn't look right in the original code either.

We need the following scenarios to be tested:
1) Site admin can view all users
2) Student can view their own records.
3) Student cannot view another students records by changing the url
4) A parent role can see their students records.

I don't think the current implentation was designed to allow teachers to view the details from users within their course but it might be worth verifying that.